runtime/vm/resolvesupport.cpp

```diff
@@ -872,8 +875,26 @@ resolveInstanceFieldRefInto(J9VMThread *vmStruct, J9Method *method, J9ConstantPo
 		nameAndSig = J9ROMFIELDREF_NAMEANDSIGNATURE(romFieldRef);
 		name = J9ROMNAMEANDSIGNATURE_NAME(nameAndSig);
 		signature = J9ROMNAMEANDSIGNATURE_SIGNATURE(nameAndSig);
-		fieldOffset = instanceFieldOffsetWithSourceClass(vmStruct, resolvedClass, J9UTF8_DATA(name), J9UTF8_LENGTH(name), J9UTF8_DATA(signature), J9UTF8_LENGTH(signature), &definingClass, (UDATA *)&field, lookupOptions, classFromCP);
-		
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+		if (J9_IS_J9CLASS_VALUETYPE(resolvedClass)) {
+			if ('Q' == J9UTF8_DATA(signature)[0]) {
+				J9FlattenedClassCache *flattenedClassCache = resolvedClass->flattenedClassCache;
+				fieldIndex = findIndexInFlattenedClassCache(flattenedClassCache, nameAndSig);
+				Assert_VM_false(fieldIndex == UDATA_MAX);
+				J9FlattenedClassCacheEntry * flattenedClassCacheEntry = J9_VM_FCC_ENTRY_FROM_FCC(flattenedClassCache, fieldIndex);
+				fieldOffset = flattenedClassCacheEntry->offset;
+				if (fieldOffset != -1) {
+					definingClass = resolvedClass;
+					field = flattenedClassCacheEntry->field;
+					flattenableClass = flattenedClassCacheEntry->clazz;
+					fccEntryFieldSet = true;
```

Comment:
this can be reversed to `fccEntryFieldNotSet` since this is the case you are checking for